### DISCUSSION TITLE
替换Country.mmdb需要添加策略吗，最新探测功能内存推荐多大合适

### DISCUSSION THREAD

**Discussion by gukey** [kkkgo/PaoPaoGateWay#6] [问和答]
替换Country.mmdb需要加规则么，这个：
```
rules:
  - GEOIP,PRIVATE,policy,no-resolve
  - GEOIP,FACEBOOK,policy
  - GEOIP,CN,policy,no-resolve
```
用 `https://github.com/Loyalsoldier/geoip` 的库；
如果替换为无精简版本GeoIP库，对解析性能会有影响吗，另外就是最新嗅探特性虚拟机多大内存好一些；

是否考虑开启下Discussions，感觉找问题简单些；

**Comment by kkkgo** [comment]
其实`Country.mmdb`默认内置是两个作用：
1 . clash默认启动需要Country.mmdb，不然会尝试下载导致报错；
2. 某些订阅默认的Rule里面有CN的规则，可有可无的给他加上吧

因为PPGW的工作角色是代理网关，分流实际上是由DNS和静态路由来控制，如果你没有自定义rule的话他是跑在全局模式下的，根本用不到GEOIP里面的数据，所以GEOIP数据并不怎么重要（这一点在文档的模式方面的内容也有说明，集成到ISO的GEOIP不能更新，所以意义不是很大），GEOIP是适合那些有很多线路，然后不同的目的IP走不同线路的复杂规则的需求，但这种需求其实单独开个docker跑，开个socks5代理给ppgw会更灵活，数据更新也方便。至于对解析性能会有影响吗？其实取决于你的规则是否复杂。  
内存的话其实512MB应该很够用了，企业环境1G也比较宽裕了，毕竟系统本身占用的比较少。如果你正常用挺流畅的话其实不需要嗅探。

**Reply by gukey** [reply]
好的，新版本docker 是不是默认没有包含 ` Country.mmdb `文件，我直接
```
docker pull sliamb/ppgwiso
docker run --rm -v .:/data sliamb/ppgwiso
```
 打包，镜像无法访问油管这些网站，我又手动下载了 ` Country.mmdb `文件，上传到当前目录，重新打包可以用了；

**Reply by kkkgo** [reply]
是数据更换出错了，现在应该好了。

**Reply by gukey** [reply]
这会重新打包试了下，不用上传 `Country.mmdb` 也可以用了。
